### PR TITLE
Use a GitHub token if provided

### PR DESCRIPTION
**Pull request by ChlodAlejandro**
Fixes https://github.com/browser-actions/setup-geckodriver/issues/7, presumably fixes https://github.com/browser-actions/setup-geckodriver/issues/6 (error is inconclusive, but 403 is returned by GitHub whenever the rate limit is hit).

This enables a user to provide a GitHub token using the token input. This avoids hitting GitHub rate limits, especially for the Windows and macOS runners, by allowing the user to pass the runner GitHub token, authorizing requests and allowing a much higher rate limit. Since rate limits are per-IP, this will also prevent users from hitting the rate limit if they're using a GitHub-provided runner (as runners may appear under the same IP address), even if they only use this action once (the intermittent failure issue).

Supersedes #9, as a force-push was made.

### CODE REVIEW

**Comment by jonowo** [comment]
Hey @ChlodAlejandro, can you publish your fork's action to the GitHub Marketplace? I don't think this project is being maintained.

**Comment by ChlodAlejandro** [comment]
@jonowo I'm not really keen on publishing a fork to the Marketplace without approval from the repo maintainer(s). In any case, I've emailed @ueokande for a review, hopefully they can reply soon.

For now, you can use the following in your workflow files. Make sure to switch over to this action whenever this gets merged.
```yml
      - name: Install geckodriver
        uses: ChlodAlejandro/setup-geckodriver@latest
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
```

**Review by ueokande — APPROVED** [review]
thank you!

**Comment by m4rc1e** [comment]
Thank you both!

**Pull request merged**
